server/user.js

```diff
@@ -0,0 +1,191 @@
+/**
+ * @fileoverview user function handler
+ */
+var fs = require('fs');
+var assert = require('assert');
+
+var log = require('./log.js');
+var utils = require('./utils.js');
+
+/** @type {user} */
+module.exports = user;
+
+/**
+ * @constructor
+ */
+function user() {}
+
+/** @enum {string} */
+user.QueryType = {
+  SIGNUP: 'sign-up',
+  SIGNIN: 'sign-in'
+};
+
+/**
+ * Name of user information file.
+ * @type {string}
+ */
+user.userInfoFile = 'userInfo.txt';
+
+/**
+ * Expire time of cookie.
+ * @const {number}
+ */
+user.cookieExpireTime = 24 * 60 * 60 * 1000;
+
+/**
+ * @typedef {{
+ *   email: (string|undefined),
+ *   username: string,
+ *   password: string,
+ *   confirmed: (boolean|undefined)
+ * }}
+ */
+user.userInfo;
```

Comment:
Redundant variable naming as you have repeated "user". user.Info is
sufficient

On Sunday, March 20, 2016, Liana Yang <notifications@github.com> wrote:

> In server/user.js
> <https://github.com/ViDA-NYU/genotet/pull/145#discussion_r56759318>:
>
> > +
> > +/**
> > + * Expire time of cookie.
> > + * @const {number}
> > + */
> > +user.cookieExpireTime = 24 * 60 * 60 * 1000;
> > +
> > +/**
> > + * @typedef {{
> > + *   email: (string|undefined),
> > + *   username: string,
> > + *   password: string,
> > + *   confirmed: (boolean|undefined)
> > + * }}
> > + */
> > +user.userInfo;
>
> Why redundant? When user information object is used as input, we should
> write the @param <https://github.com/param> as a name for the type of
> object.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/ViDA-NYU/genotet/pull/145/files/83dd548aa3090df14c14b115859530f16cb2a2a5#r56759318>
>